### PR TITLE
SearchParamsSummary component

### PR DESCRIPTION
**Pull request by lukewalczak**

### CODE REVIEW

File: src/utils/DateUtils/DateFormatter.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    day: 'numeric',
+  }).format(date);
+}
+
+function custom(date: Date, config: FormatterConfig) {
+  return Intl.DateTimeFormat(DEVICE_LOCALE, {
+    ...config,
+    timeZone: 'UTC', // this is very important!
+  }).format(date);
+}
```
Comment (FilipMessa) [inline src/utils/DateUtils/DateFormatter.js]: What about something like?
```
createDateFormater = config => date => {
  return function dateFromtater(date) {
    return Intl.DateTimeFormat(DEVICE_LOCALE, {
      ...config,
      timeZone: "UTC" // this is very important!
    }).format(date);
  };
};

const birthday = createDateFormater({
  year: "numeric",
  month: "numeric",
  day: "numeric"
});

```

Comment (tbergq) [reply src/utils/DateUtils/DateFormatter.js]: Isn't the Intl somehow patched in our mobile repo? Because of some troubles with android? cc @ferrannp

Comment (lukewalczak) [reply src/utils/DateUtils/DateFormatter.js]: It's patched there as well, on iOS works fine. Will have to check Android.

Comment (tbergq) [reply src/utils/DateUtils/DateFormatter.js]: https://github.com/kiwicom/mobile/blob/master/scripts/configureApplication.js#L87

Comment (lukewalczak) [reply src/utils/DateUtils/DateFormatter.js]: https://github.com/kiwicom/universal-components/blob/e768c49fb9d0ebffe4b82c966c3eaf531aad1743/scripts/postInstall.js

Comment (RobinCsl) [reply src/utils/DateUtils/DateFormatter.js]: I would be happy to have @FilipMessa's proposal to make the code more succinct. At the end of the day, it's achieving the same.

Comment (lukewalczak) [reply src/utils/DateUtils/DateFormatter.js]: Ok, I can rewrite it, it's not an issue. I was just thinking we would like to have the same implementation in two repos. :)

File: scripts/postInstall.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// @flow strict
+
+const fs = require('fs');
+const path = require('path');
+
+const log = message => console.log(`➤ ${message}`); // eslint-disable-line no-console
+
+log('Patching intl package');
```
Comment (RobinCsl) [inline scripts/postInstall.js]: Maybe could you add a comment linking to a GitHub issue explaining why this patch is necessary?

File: src/SearchParamsSummary/SearchParamsSummary.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    default:
+      icon = 'flight-return';
+  }
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.headerLeftcontainer}>
+        <View style={styles.citiesContainer}>
+          <Text style={styles.departureCity}>
+            {(departure && departure.city) || ''}
```
Comment (RobinCsl) [inline src/SearchParamsSummary/SearchParamsSummary.js]: I think we are using optional chaining with `??` operator too.

File: src/SearchParamsSummary/__tests__/index.test.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import Text from '../../Text';
+import { Icon } from '../../Icon';
+import { SearchParamsSummary } from '../index';
+import AdaptableBadge from '../AdaptableBadge';
+
+jest.mock('../../utils/DateUtils/DateFormatter', () => ({
+  custom: () => 'Oct 10',
+}));
+
+const renderHeaderComponent = (tripType, component) => {
```
Comment (RobinCsl) [inline src/SearchParamsSummary/__tests__/index.test.js]:
```suggestion
const renderSearchParamsSummary = (tripType, component) => {
```

File: src/SearchParamsSummary/__tests__/index.test.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      arrival={{ city: 'Prague', date: '2018-12-12' }}
+    />
+  );
+
+  return getByType(component);
+};
+
+describe('SearchParamsSummary', () => {
+  it('should have icon name "flight-direct" when trip type is "OneWay"', () => {
+    const icon = renderHeaderComponent('OneWay', Icon);
```
Comment (RobinCsl) [inline src/SearchParamsSummary/__tests__/index.test.js]:
```suggestion
    const icon = renderSearchParamsSummary('OneWay', Icon);
```

File: src/SearchParamsSummary/__tests__/index.test.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+    expect(icon.props.name).toBe('flight-direct');
+  });
+
+  it('should have icon name "flight-multicity" when trip type is "MultiCity"', () => {
+    const icon = renderHeaderComponent('MultiCity', Icon);
```
Comment (RobinCsl) [inline src/SearchParamsSummary/__tests__/index.test.js]:
```suggestion
    const icon = renderSearchParamsSummary('MultiCity', Icon);
```

File: src/SearchParamsSummary/__tests__/index.test.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+    expect(icon.props.name).toBe('flight-multicity');
+  });
+
+  it('should have icon name "flight-direct" when trip type is "Return"', () => {
+    const icon = renderHeaderComponent('Return', Icon);
```
Comment (RobinCsl) [inline src/SearchParamsSummary/__tests__/index.test.js]:
```suggestion
    const icon = renderSearchParamsSummary('Return', Icon);
```

**Pull request merged**